### PR TITLE
Update release schedule for v1.31

### PR DESCRIPTION
**Pull request by Princesso**
Add an entry for v1.31 on the releases schedule, as described [here](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/docs/Release-Timeline.md#update-releases-page-the-week-before-the-release)

cc: @katcosgrove @saschagrunert @drewhagen @salaxander @sftim

### CODE REVIEW

**Comment by saschagrunert** [comment]
Let’s do this after v1.31.0 to not confuse users

/hold

**Comment by sftim** [comment]
@saschagrunert you're wrong to hold this. We need this change to land _in order_ to green light the release.

**Review by saschagrunert — APPROVED** [review]
/unhold

interesting, I wasn’t aware of this.

**Review by sftim — COMMENTED** [review]
LGTM

Would we also like to update `data/releases/eol.yaml` (and retitle this PR to match)?

**Comment by sftim** [comment]
During release, we merge _dev-1.31_ into _main_, and we'd like the website to then correctly show that a v1.31 exists (and is available to download, etc).

There is now code to handle the situation where updating the release schedule has been missed but at least IMO we should hold the release if the update hasn't happened (imagine we noticed on release day). And then very quickly we'd do a PR like this one and unhold the release.

**Comment by sftim** [comment]
Here's a [preview](https://deploy-preview-47338--kubernetes-io-main-staging.netlify.app/releases/). The preview incorrectly states that v1.30 is the latest release at the top of the page, but https://github.com/kubernetes/website/pull/47319 would fix that if merged.

**Comment by saschagrunert** [comment]
Missed that this PR is for dev-1.31, now everything makes sense

**Pull request merged**
